lighthouse-core/gather/computed/dependency-graph/estimator/estimator.js

```diff
@@ -221,9 +288,17 @@ class Estimator {
    * @param {number} totalElapsedTime
    */
   _updateProgressMadeInTimePeriod(node, timePeriodLength, totalElapsedTime) {
+    const timingData = this._nodeTiming.get(node);
+    const isFinished = timingData.estimatedTimeElapsed === timePeriodLength;
```

Comment:
feels like one and only one node should be finishing per `estimateWithDetails` loop to make reasoning about progress easier (if thinking of it as a sweep line), but you may be thinking differently

one option: `_findNextNodeCompletionTime` could be `_findNextCompletedNode`, and `_updateProgressMadeInTimePeriod` could finish only that one and use its timing for `totalElapsedTime`

Reply:
because the network logic happens in essentially round trip blocks rather than on a packet simulation level, it's possible to have multiple requests finishing at exactly the same time and get updated all at once and this simplified things to not have to worry about 0 elapsed time

if you think that's confusing though I'm open to a future refactor since that would line up nicely with information stored on each node

Reply:
> if you think that's confusing though I'm open to a future refactor since that would line up nicely with information stored on each node

yeah, I think it would generally be easier to reason about if there was only one node finishing at any one point (where that point is defined by actual time + some ordering of the nodes)